### PR TITLE
test: enable sync cloud storage e2e test

### PR DESCRIPTION
**Pull request by Jonas1312**
enables the e2e sync cloud storage test

we added some data integration id in github secret KILI_TEST_DATA_INTEGRATION_ID:
```python
    """
    {
        "LTS": {
            "AWS": ["data_integration_id_1", "data_integration_id_2"],
            "GCP": ["data_integration_id_1", "data_integration_id_2"],
            "Azure": ["data_integration_id_1", "data_integration_id_2"],
        },
        "STAGING": {
            "AWS": ["data_integration_id_1", "data_integration_id_2"],
            ...
        },
        ...
    }
    """
```

also add retries to check when validateDataDifferences is finished. If we don't, the query will return before the validateDataDifferences is really effective (mutation + query problem)

e2e: https://github.com/kili-technology/kili-python-sdk/actions/runs/4385846340

```text
tests/e2e/test_cloud_storage.py::test_e2e_synchronize_cloud_storage_connection[STAGING-AWS-clf2****************wjm] Data integration used: clf****************wjm

  0%|          | 0/1 [00:00<?, ?it/s]
100%|██████████| 1/1 [00:00<00:00,  2.49it/s]
100%|██████████| 1/1 [00:00<00:00,  2.49it/s]
Synchronizing data connection: c****************hv4
Verifying that the data connection differences have been computed
Currently 0 asset(s) imported from the data connection.
Found 6066 difference(s): 6066 assets to add, 0 assets to remove.
Added 6066 file(s).

Data connection: ***'dataDifferencesSummary': ***'added': 6066, 'removed': 0, 'total': 6066***, 'id': 'clf****************l4hv4', 'isChecking': True, 'isApplyingDataDifferences': False, 'lastChecked': '2023-03-10T15:13:48.111Z', 'numberOfAssets': 0, 'selectedFolders': ['animals', 'birds'], 'dataIntegrationId': 'clf2ej****************wjm'***
PASSED
```

if we test staging key on lts endpoint it'll output:
```
tests/e2e/test_cloud_storage.py::test_e2e_synchronize_cloud_storage_connection[STAGING-AWS-clf2****************cwjm] SKIPPED (Skipping test because it is not the right endpoint.)
``

### CODE REVIEW

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: tests/mutations/test_sync_data_connection.py [hunk 1]
```diff
@@ -39,82 +46,92 @@ def __call__(self, auth, data_connection_id: str, fields: List[str]) -> Dict[str
...
-            assert args[1] == {"projectID": "my_project_id"}
-
-        def __iter__(self):
-            return self
-
-        def __next__(self):
-            return {"data": {"isChecking": False}}
-
     kili = MutationsDataConnection(auth=MagicMock(client=mocked_graphql_client))
-    mocked_graphql_client.subscribe = MagicMock(side_effect=BackendSubscriptionEventGenerator)
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply tests/mutations/test_sync_data_connection.py:117]: I'm honestly not sure if we should avoid using MagicMock. pytest-mock is just a wrapper on top of unittest.mock package, so I don't see the difference

**Review by PierreLeveau — APPROVED** [review]

File: src/kili/services/data_connection/__init__.py [hunk 2]
```diff
@@ -160,7 +190,7 @@ def synchronize_data_connection(
             )
 
     if added > 0:
-        data_connection = validate_data_differences(auth, "ADD", data_connection_id)
+        validate_data_differences(auth, "ADD", data_connection)
```
Comment (theodu) [inline src/kili/services/data_connection/__init__.py:193]: ADD and REMOVED are necessarily done in two different calls ?

**Review by theodu — APPROVED** [review]

File: src/kili/services/data_connection/__init__.py (hunk 2 above)
Comment (Jonas1312) [reply src/kili/services/data_connection/__init__.py:193]: yes I think:

```graphql
  validateDataDifferences(where: ValidateDataDifferencesWhere!, processingParameters: String): DataConnection

input ValidateDataDifferencesWhere {
  connectionId: ID!
  type: DataDifferenceType!
}

enum DataDifferenceType {
  ADD
  REMOVE
}
```

Comment (theodu) [reply src/kili/services/data_connection/__init__.py:193]: indeed!

**Pull request merged**